databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -475,6 +546,95 @@ impl proto::val_server::Val for broker::DataBroker {
     }
 }
 
+async fn validate_entry_update(
+    broker: &AuthorizedAccess<'_, '_>,
+    request: &EntryUpdate,
+    id: i32,
+) -> Result<Pair<i32, broker::EntryUpdate>, Status> {
+    let entry = &request.entry.clone().unwrap();
+
+    let fields = HashSet::<proto::Field>::from_iter(request.fields.iter().filter_map(
+        |id| proto::Field::from_i32(*id), // Ignore unknown fields for now
+    ));
+
+    if entry.actuator_target.is_some() {
+        if let Some(metadata) = broker.get_metadata(id).await {
+            if metadata.entry_type != broker::EntryType::Actuator {
+                return Err(tonic::Status::invalid_argument(
+                    "Tried to set a target value for a non-actuator. Non-actuators have no target value.".to_string(),
+                ));
+            }
+        }
+    }
+
+    let entry = match &request.entry {
+        Some(entry) => entry,
+        None => return Err(tonic::Status::invalid_argument("Empty entry".to_string())),
+    };
+
+    debug!("Setting fields: {:?}", fields);
+    let update = broker::EntryUpdate::from_proto_entry_and_fields(entry, fields);
+
+    let pair = Pair {
+        first: id,
+        second: update,
+    };
+    Ok(pair)
+}
+
+fn convert_to_data_entry_error(path: &String, error: &broker::UpdateError) -> DataEntryError {
```

Comment:
would be nice to add a generic one for converting these maybe moving this into conversion.rs as well. https://github.com/eclipse-kuksa/kuksa-databroker/blob/82513eaf486dbc4e4e60fad0048b3cf8a10bfd8f/databroker/src/grpc/kuksa_val_v1/val.rs#L453 because here we do it in place.

Reply:
These are custom errors, which are returned and not propagated back to the client using the DataEntryError, therefore I can not move them into the convert_to_data_entry_error method